modules/core/src/effects/effect-preparator.js

```diff
@@ -0,0 +1,10 @@
+export default class EffectPreparator {
```

Comment:
Add comments to document the purpose of this class? Can this just be a part of the `EffectManager`?

Reply:
+1 for putting this into `EffectManager` unless that class becomes too big.

Reply:
Added comments to explain the purpose of this class, I don't want EffectManager to know layers and other resources which is necessary for effects , so prefer to add this class.

Reply:
Who will own this class then? Can we just move this code into the parent class until it gets too big?

Reply:
DeckRenderer will own this class, DeckRenderer focuses on pass assembly, it only cares about the final output of effect modules, don't want it knows the effect's APIs and some internal relations.